worker-store-fs/src/main/java/com/hpe/caf/worker/datastore/fs/FileSystemDataStore.java

```diff
@@ -223,7 +237,25 @@ public void close() throws IOException
     @Override
     public HealthResult healthCheck()
     {
-        return HealthResult.RESULT_HEALTHY;
+        final Future<HealthResult> healthcheckFuture = healthcheckExecutor.submit(healthcheck);
+        try {
+            return healthcheckFuture.get(healthcheckTimeout.getSeconds(), TimeUnit.SECONDS);
+        } catch (final TimeoutException e) {
+            healthcheckFuture.cancel(true);
+            return new HealthResult(
+                HealthStatus.UNHEALTHY,
+                String.format("Timeout after %s seconds trying to access data store directory %s",
+                              healthcheckTimeout.getSeconds(),
+                              dataStorePath.toString()));
+        } catch (final Exception e) {
```

Comment:
You've changed this from `} catch (final InterruptedException | ExecutionException e) {` to `} catch (final Exception e) {`.  Are you trying to catch unchecked exceptions as well or why have you done this?

Reply:
Unless you had a reason for changing this I think you had it right before:
```suggestion
        } catch (final InterruptedException | ExecutionException e) {
```

Reply:
Yes thanks, I mistakenly changed it to Exception so that it would catch the IOException thrown from the healthcheck as well, but I can see now that if an IOException is thrown, it will be wrapped in an ExecutionException. Reverted.